pallets/xcm-transactor/src/lib.rs

```diff
@@ -508,6 +512,15 @@ pub mod pallet {
 				total_weight.clone(),
 			)?;
 
+			// If refund is true, the appendix instruction will be a deposit back to the sovereign
+			let appendix: Option<Vec<Instruction<()>>> = if refund {
```

Comment:
```suggestion
			let appendix = refund.then(|| {
```

It's more idiomatic to use then instead: https://doc.rust-lang.org/stable/std/primitive.bool.html#method.then

Reply:
In that case, I will need to perform something like:
```rust
let appendix = refund.then(|| {
	let deposit_appendix =
		Self::deposit_instruction(T::SelfLocation::get(), &dest, 1u32);
	vec![RefundSurplus, deposit_appendix.unwrap_or(RefundSurplus)]
});
```
This is needed because the `deposit_appendix()` function returns a `Result`, and I can't make use of the mark operator `?` inside the scope of the closure of `then()`, so we are enforced to unwrap the value or put a default one. In this case, if there is an error inside `deposit_appendix()` function the appendix will result in two `RefundSurplus` instructions and nothing will be refunded. I think the `unwrap_or(RefundSurplus)` is needed to avoid panics in the runtime that could appear if we were to use only `unwrap()`. 

Is this okay for you or do you have any other suggestions?

Reply:
You can return a result in the closure then use transpose : https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.transpose

```rust
let appendix = refund.then(|| {
	Ok(vec![RefundSurplus, Self::deposit_instruction(T::SelfLocation::get(), &dest, 1u32)?])
}).transpose()?;
```

Reply:
Works fine, I also had to add the return type to the closure because otherwise it wasn't recognizing the `DispatchError` type thrown by the `deposit_appendix()` function. Thanks :)